Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/FirebaseCloudMessaging/FirebaseCloudMessagingConnector.cs

```diff
@@ -69,22 +71,27 @@ public void ValidateRegistration(PnsRegistration registration)
             throw new InfrastructureException(InfrastructureErrors.InvalidPushNotificationConfiguration(_options.GetSupportedAppIds()));
     }
 
-    private List<SendResult> MapResponse(BatchResponse batchResponse, IReadOnlyList<DeviceId> devices)
+    private SendResults MapResponse(BatchResponse batchResponse, IReadOnlyList<DeviceId> devices, SendResults sendResults)
     {
-        var sendResults = new List<SendResult>();
         for (var index = 0; index < batchResponse.Responses.Count; index++)
         {
             var response = batchResponse.Responses[index];
             var deviceId = devices[index];
             if (response.IsSuccess)
             {
-                sendResults.Add(SendResult.Success());
+                sendResults.AddSuccess(deviceId);
             }
             else
             {
-                sendResults.Add(response.Exception.MessagingErrorCode is MessagingErrorCode.InvalidArgument or MessagingErrorCode.Unregistered
-                    ? SendResult.Failure(deviceId, SendResult.FailureReason.InvalidHandle)
-                    : SendResult.Failure(deviceId, SendResult.FailureReason.Unexpected, response.Exception.Message));
+                switch (response.Exception.MessagingErrorCode)
+                {
+                    case MessagingErrorCode.InvalidArgument or MessagingErrorCode.Unregistered:
```

Comment:
According to the [documentation](https://firebase.google.com/docs/cloud-messaging/manage-tokens#detect-invalid-token-responses-from-the-fcm-backend), there are two cases for an invalid token. Why didn't you consider `INVALID_ARGUMENT` here?

Reply:
I think you didn't read the code properly, is says ` MessagingErrorCode.InvalidArgument or MessagingErrorCode.Unregistered` 🤔

Reply:
Whoops, you're right. My bad